crates/matrix-sdk-base/src/store/mod.rs

```diff
@@ -676,14 +704,14 @@ impl StoreConfig {
     ///
     /// The crypto store must be opened before being set.
     #[cfg(feature = "e2e-encryption")]
-    pub fn crypto_store(mut self, store: impl CryptoStore + 'static) -> Self {
-        self.crypto_store = Some(Arc::new(store));
+    pub fn crypto_store(mut self, store: impl IntoCryptoStore) -> Self {
```

Comment:
I wonder if it's clear enough that this type is the same as `CryptoStore`. I guess it becomes clear if you follow the link to the trait definition.

Reply:
Yeah, if you don't copy-paste from an example, you'll have to click one more link most likely. That seems fine to me.